org-caldav.el

```diff
@@ -1189,8 +1197,11 @@ is on s-expression."
   (when (eq backend 'icalendar)
     (org-map-entries
      (lambda ()
-       (let ((pt (save-excursion (apply 'org-agenda-skip-entry-if org-caldav-skip-conditions))))
-         (when pt (delete-region (point) (- pt 1))))))))
+       (let ((pt (save-excursion (apply 'org-agenda-skip-entry-if org-caldav-skip-conditions)))
+             (ts (when org-caldav-days-in-past (* (abs org-caldav-days-in-past) -1)))
+             (stamp (org-entry-get nil "TIMESTAMP" t)))
+         (when (or pt (and stamp (> ts (org-time-stamp-to-now stamp))))
+           (delete-region (point) (or (org-end-of-subtree t)))))))))
```

Comment:
Not sure why there is this or. The following code is equivalent, and simpler:
```suggestion
           (delete-region (point) (org-end-of-subtree t))))))))
```

Reply:
Good catch, It probably was introduced by me while rebasing.
Thanks for the review!